libs/langchain/langchain/chat_models/fireworks.py

```diff
@@ -0,0 +1,261 @@
+import fireworks
+import fireworks.client
+from langchain.utils.env import get_from_dict_or_env
+from pydantic import root_validator
+from typing import Any, Callable, Dict, Iterator, List, Mapping, Optional, Tuple, Union
+from langchain.adapters.openai import convert_message_to_dict
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain.chat_models.base import BaseChatModel
+from langchain.llms.base import create_base_retry_decorator
+from langchain.schema.messages import (
+    AIMessage,
+    AIMessageChunk,
+    BaseMessage,
+    BaseMessageChunk,
+    ChatMessage,
+    FunctionMessage,
+    HumanMessage,
+    SystemMessage,
+    AIMessageChunk,
+    BaseMessage,
+    BaseMessageChunk,
+    ChatMessageChunk,
+    FunctionMessageChunk,
+    HumanMessageChunk,
+    SystemMessageChunk,
+)
+from langchain.schema.output import ChatGeneration, ChatGenerationChunk, ChatResult
+
+
+def _convert_delta_to_message_chunk(
```

Comment:
Not sure what's the convention in the langchain codebase, but in general, it would be good to add some comments.